src/app/components/challenge/challengephases/editphasemodal/editphasemodal.component.ts

```diff
@@ -0,0 +1,199 @@
+import { ViewChildren, QueryList, Component, Input, OnInit } from '@angular/core';
+import { GlobalService } from '../../../../services/global.service';
+import { InputComponent } from '../../../utility/input/input.component';
+
+@Component({
+  selector: 'app-editphasemodal',
+  templateUrl: './editphasemodal.component.html',
+  styleUrls: ['./editphasemodal.component.scss']
+})
+export class EditphasemodalComponent implements OnInit {
+
+  /**
+   * Input parameters object
+   */
+  @Input() params: any;
+
+  /**
+   * Modal title
+   */
+  title = 'Edit Challenge Phase Details';
+
+  /**
+   * Label for the editor field i.e. challenge phase description
+   */
+  label = '';
+
+  /**
+   * Challenge phase name
+   */
+  name = '';
+
+  /**
+   * Challenge phase description
+   */
+  description = '';
+
+  /**
+   * Challenge phase start date
+   */
+  startDate = '';
+
+  /**
+   * Challenge phase end date
+   */
+  endDate = '';
+
+  /**
+   * Challenge phase max submissions per day
+   */
+  maxSubmissionsPerDay = '';
+
+  /**
+   * Challenge phase max submissions per month
+   */
+  maxSubmissionsPerMonth = '';
+
+  /**
+   * Challenge phase max submissions
+   */
+  maxSubmissions = '';
```

Comment:
Can we do this?
```suggestion
  maxSubmissions: number;
```
(And similarly in other places)